From 59faec3a09febbde89491f321614324b736faf53 Mon Sep 17 00:00:00 2001
From: Xiaolei Wang <xiaolei.wang@windriver.com>
Date: Fri, 24 Sep 2021 09:20:58 +0800
Subject: [PATCH] nxp-imx8/generate-graphic-layer.sh: add machine learning 
 override convert

Issue:  LINUXEXEC-16054

apply a override convert patch when run script.

Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 recipes-devtools/armnn-swig/armnn-swig.inc       |  6 +++---
 .../flatbuffers/flatbuffers-native_1.11.0.bb     |  2 +-
 .../flatbuffers/flatbuffers_1.11.0.bb            |  4 ++--
 .../arm-compute-library_git-imx.bb               |  4 ++--
 .../armnn/armnn-tensorflow_1.15.0.bb             |  2 +-
 recipes-libraries/armnn/armnn_20.08.bb           | 10 +++++-----
 recipes-libraries/nn-imx/nn-imx_1.1.9.bb         |  4 ++--
 .../onnxruntime/onnxruntime_1.5.3.bb             |  8 ++++----
 .../tensorflow-lite/tensorflow-lite_2.4.0.bb     | 16 ++++++++--------
 9 files changed, 28 insertions(+), 28 deletions(-)

diff --git a/recipes-devtools/armnn-swig/armnn-swig.inc b/recipes-devtools/armnn-swig/armnn-swig.inc
index ca064e6..1afcb0f 100644
--- a/recipes-devtools/armnn-swig/armnn-swig.inc
+++ b/recipes-devtools/armnn-swig/armnn-swig.inc
@@ -50,11 +50,11 @@ BBCLASSEXTEND = "native nativesdk"
 
 # necessary together with bison dependency until a new upstream version after
 # 3.0.12 includes 0001-Fix-generated-code-for-constant-expressions-containi.patch
-do_configure_append() {
+do_configure:append() {
     mkdir -p ${B}/Source/CParse
 }
 
-do_install_append_class-nativesdk() {
+do_install:append:class-nativesdk() {
     cd ${D}${bindir}
     ln -s swig swig4.0
 }
@@ -63,7 +63,7 @@ def swiglib_relpath(d):
     swiglib = d.getVar('datadir') + "/" + 'swig' + "/" + d.getVar('PV')
     return os.path.relpath(swiglib, d.getVar('bindir'))
 
-do_install_append_class-native() {
+do_install:append:class-native() {
     create_wrapper ${D}${bindir}/swig SWIG_LIB='`dirname $''realpath`'/${@swiglib_relpath(d)}
 }
 FILES:${PN}-dev += "${datadir_native}/*"
diff --git a/recipes-devtools/flatbuffers/flatbuffers-native_1.11.0.bb b/recipes-devtools/flatbuffers/flatbuffers-native_1.11.0.bb
index 242fc51..e11058d 100644
--- a/recipes-devtools/flatbuffers/flatbuffers-native_1.11.0.bb
+++ b/recipes-devtools/flatbuffers/flatbuffers-native_1.11.0.bb
@@ -32,6 +32,6 @@ do_install() {
     install -m 0755 ${B}/flatc ${STAGING_BINDIR_NATIVE}
 }
 
-FILES_${PN} += "${bindir}"
+FILES:${PN} += "${bindir}"
 
 #BBCLASSEXTEND = "native nativesdk"
diff --git a/recipes-devtools/flatbuffers/flatbuffers_1.11.0.bb b/recipes-devtools/flatbuffers/flatbuffers_1.11.0.bb
index fc838e4..ac7dd00 100644
--- a/recipes-devtools/flatbuffers/flatbuffers_1.11.0.bb
+++ b/recipes-devtools/flatbuffers/flatbuffers_1.11.0.bb
@@ -26,9 +26,9 @@ inherit cmake
 
 S = "${WORKDIR}/git"
 
-do_install_append() {
+do_install:append() {
     install -d ${D}${bindir}
     install -m 0755 ${B}/flatc ${D}${bindir}
 }
 
-FILES_${PN} += "${libdir}"
+FILES:${PN} += "${libdir}"
diff --git a/recipes-libraries/arm-compute-library/arm-compute-library_git-imx.bb b/recipes-libraries/arm-compute-library/arm-compute-library_git-imx.bb
index e39a45d..f2b5d9d 100644
--- a/recipes-libraries/arm-compute-library/arm-compute-library_git-imx.bb
+++ b/recipes-libraries/arm-compute-library/arm-compute-library_git-imx.bb
@@ -101,7 +101,7 @@ INHIBIT_PACKAGE_DEBUG_SPLIT = "1"
 SOLIBS = ".so"
 FILES_SOLIBSDEV = ""
 
-RDEPENDS_${PN} = "bash"
+RDEPENDS:${PN} = "bash"
 
 # We support i.MX8 only (for now)
 COMPATIBLE_MACHINE = "(mx8)"
diff --git a/recipes-libraries/armnn/armnn-tensorflow_1.15.0.bb b/recipes-libraries/armnn/armnn-tensorflow_1.15.0.bb
index 6043bdd..e916fa3 100644
--- a/recipes-libraries/armnn/armnn-tensorflow_1.15.0.bb
+++ b/recipes-libraries/armnn/armnn-tensorflow_1.15.0.bb
@@ -33,4 +33,4 @@ do_install() {
     done
 }
 
-FILES_${PN} += "${datadir}"
+FILES:${PN} += "${datadir}"
diff --git a/recipes-libraries/armnn/armnn_20.08.bb b/recipes-libraries/armnn/armnn_20.08.bb
index 88c1848..83baec2 100644
--- a/recipes-libraries/armnn/armnn_20.08.bb
+++ b/recipes-libraries/armnn/armnn_20.08.bb
@@ -29,7 +29,7 @@ RDEPENDS_MX8       = ""
 RDEPENDS_MX8_mx8   = "nn-imx"
 RDEPENDS_MX8_mx8mm = ""
 RDEPENDS_MX8_mx8mnlite = ""
-RDEPENDS_${PN}   = " \
+RDEPENDS:${PN}   = " \
     arm-compute-library \
     protobuf \
     boost \
@@ -62,7 +62,7 @@ EXTRA_OECMAKE += " \
 ARMNN_INSTALL_DIR = "${bindir}/${P}"
 PYARMNN_INSTALL_DIR = "${ARMNN_INSTALL_DIR}/pyarmnn"
 
-do_compile_append() {
+do_compile:append() {
     if ${@bb.utils.contains('PACKAGECONFIG', 'pyarmnn', 'true', 'false', d)}; then
         # copy required to link against pyarmnn wrappers
         # due to a bug in python/setuptools an explicit path cannot be set 
@@ -85,7 +85,7 @@ do_compile_append() {
     fi
 }
 
-do_install_append() {
+do_install:append() {
     if ${@bb.utils.contains('PACKAGECONFIG', 'tests', 'true', 'false', d)}; then
         install -d ${D}${ARMNN_INSTALL_DIR}
         CP_ARGS="-Prf --preserve=mode,timestamps --no-preserve=ownership"        
@@ -110,8 +110,8 @@ do_install_append() {
 CXXFLAGS += "-fopenmp -Wno-uninitialized"
 LIBS += "-larmpl_lp64_mp"
 
-FILES_${PN} += "${libdir}/python*"
-FILES_${PN} += "${PYARMNN_INSTALL_DIR}/examples*"
+FILES:${PN} += "${libdir}/python*"
+FILES:${PN} += "${PYARMNN_INSTALL_DIR}/examples*"
 
 # We support i.MX8 only (for now)
 COMPATIBLE_MACHINE = "(mx8)"
diff --git a/recipes-libraries/nn-imx/nn-imx_1.1.9.bb b/recipes-libraries/nn-imx/nn-imx_1.1.9.bb
index b1e6f45..ecbada7 100644
--- a/recipes-libraries/nn-imx/nn-imx_1.1.9.bb
+++ b/recipes-libraries/nn-imx/nn-imx_1.1.9.bb
@@ -28,5 +28,5 @@ do_install () {
 }
 
 COMPATIBLE_MACHINE = "(mx8)"
-COMPATIBLE_MACHINE_mx8mm = "(^$)"
-COMPATIBLE_MACHINE_mx8mnlite = "(^$)"
+COMPATIBLE_MACHINE:mx8mm = "(^$)"
+COMPATIBLE_MACHINE:mx8mnlite = "(^$)"
diff --git a/recipes-libraries/onnxruntime/onnxruntime_1.5.3.bb b/recipes-libraries/onnxruntime/onnxruntime_1.5.3.bb
index a608744..4ae382f 100644
--- a/recipes-libraries/onnxruntime/onnxruntime_1.5.3.bb
+++ b/recipes-libraries/onnxruntime/onnxruntime_1.5.3.bb
@@ -36,9 +36,9 @@ EXTRA_OECMAKE += "\
 -DCMAKE_BUILD_TYPE=RelWithDebInfo \
 "
 PACKAGECONFIG_VSI_NPU       = ""
-PACKAGECONFIG_VSI_NPU_mx8   = "vsi_npu"
-PACKAGECONFIG_VSI_NPU_mx8mm = ""
-PACKAGECONFIG_VSI_NPU_mx8mnlite = ""
+PACKAGECONFIG_VSI_NPU:mx8   = "vsi_npu"
+PACKAGECONFIG_VSI_NPU:mx8mm = ""
+PACKAGECONFIG_VSI_NPU:mx8mnlite = ""
 
 PACKAGECONFIG ?= "openmp test reports sharedlib armnn eigenblas acl acl-2008 ${PACKAGECONFIG_VSI_NPU}"
 
diff --git a/recipes-libraries/tensorflow-lite/tensorflow-lite_2.4.0.bb b/recipes-libraries/tensorflow-lite/tensorflow-lite_2.4.0.bb
index 4824c92..977c7c0 100644
--- a/recipes-libraries/tensorflow-lite/tensorflow-lite_2.4.0.bb
+++ b/recipes-libraries/tensorflow-lite/tensorflow-lite_2.4.0.bb
@@ -20,7 +20,7 @@ S = "${WORKDIR}/git"
 
 EXTRA_OECMAKE = "-DTFLITE_ENABLE_XNNPACK=on -DTFLITE_ENABLE_RUY=on -DTFLITE_ENABLE_NNAPI=on  ${S}/tensorflow/lite/"
 
-do_configure_prepend(){
+do_configure:prepend(){
     export HTTP_PROXY=${http_proxy}
     export HTTPS_PROXY=${https_proxy}
     export http_proxy=${http_proxy}
@@ -35,7 +35,7 @@ do_compile:prepend () {
     sed -i 'N;30i\#include <limits>' ${S}/../build/ruy/ruy/block_map.cc
 }
 
-do_compile_append () {
+do_compile:append () {
     # build pip package
     export PYTHONPATH="${STAGING_LIBDIR_NATIVE}/${PYTHON_DIR}/site-packages"
     export PIP_BUILD_ROOT="${WORKDIR}"
@@ -87,10 +87,10 @@ do_install() {
 }
 
 RDEPENDS_MX8       = ""
-RDEPENDS_MX8_mx8   = "libnn-imx nn-imx"
-RDEPENDS_MX8_mx8mm = ""
-RDEPENDS_MX8_mx8mnlite = ""
-RDEPENDS_${PN}   = " \
+RDEPENDS_MX8:mx8   = "libnn-imx nn-imx"
+RDEPENDS_MX8:mx8mm = ""
+RDEPENDS_MX8:mx8mnlite = ""
+RDEPENDS:${PN}   = " \
     flatbuffers \
     python3 \
     python3-numpy \
@@ -102,9 +102,9 @@ SSTATE_DUPWHITELIST = "/"
 
 INHIBIT_PACKAGE_DEBUG_SPLIT = "1"
 
-INSANE_SKIP_${PN} += " \
+INSANE_SKIP:${PN} += " \
     already-stripped \
     staticdev \
 "
 
-FILES_${PN} += "${libdir}/python*"
+FILES:${PN} += "${libdir}/python*"
-- 
2.25.1

